crates/oxc_linter/src/rules/unicorn/no_thenable.rs

```diff
@@ -0,0 +1,464 @@
+                    }
+                }
+
+                // `Object.fromEntries([['then', …]])`
+                if !{
+                    !expr.optional
+                        && expr.arguments.len() == 1
+                        && matches!(
+                            expr.arguments[0],
+                            ast::Argument::Expression(Expression::ArrayExpression(_))
+                        )
+                        && match expr.callee {
+                            ast::Expression::MemberExpression(ref me) => {
+                                me.object()
+                                    .get_identifier_reference()
+                                    .map_or(false, |ident_ref| ident_ref.name == "Object")
+                                    && me.static_property_name() == Some("fromEntries")
+                                    && !me.optional()
+                            }
+                            _ => false,
+                        }
+                } {
+                } else if let ast::Argument::Expression(Expression::ArrayExpression(outer)) =
+                    &expr.arguments[0]
+                {
+                    for inner in &outer.elements {
+                        // inner item is array
+                        if let ArrayExpressionElement::Expression(Expression::ArrayExpression(
+                            inner,
+                        )) = inner
+                        {
+                            if inner.elements.len() > 0
+                                && !matches!(
+                                    inner.elements[0],
+                                    ArrayExpressionElement::SpreadElement(_)
+                                )
+                            {
+                                match inner.elements[0] {
+                                    ArrayExpressionElement::Expression(ref expr) => {
+                                        if let Some(span) = check_expression(expr, ctx) {
+                                            ctx.diagnostic(NoThenableDiagnostic::Object(span));
+                                        }
+                                    }
+                                    _ => {}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            AstKind::MemberExpression(expr) => {
+                if let Some(parent) = ctx.nodes().parent_node(node.id()) {
+                    if let Some(grandparent) = ctx.nodes().parent_node(parent.id()) {
+                        if let AstKind::AssignmentTarget(
+                            AssignmentTarget::SimpleAssignmentTarget(
+                                SimpleAssignmentTarget::MemberAssignmentTarget(target),
+                            ),
+                        ) = grandparent.kind()
+                        {
+                            if hash(&expr) == hash(target) {
```

Comment:
not sure if this is the preferred way to compare two AST 🤔

Reply:
Seems we don't need to check this. Or do you think if we can match `AssignmentTarget::SimpleAssignmentTarget` first?

Reply:
> Seems we don't need to check this. Or do you think if we can match `AssignmentTarget::SimpleAssignmentTarget` first?

Thank you for pointing it out, I will give a try